src/Host.cpp

```diff
@@ -477,11 +475,112 @@ Host::~Host()
     if (mpDockableMapWidget) {
         mpDockableMapWidget->deleteLater();
     }
-    mIsGoingDown = true;
-    mIsClosingDown = true;
     mErrorLogStream.flush();
     mErrorLogFile.close();
-    TDebug::removeHost(this);
+    // We are in the destructor - so it is not a good idea for anything we call
+    // to refer to our members, given that we want the name of the profile in
+    // the following pass it in as a value so we don't need to access the member
+    TDebug::removeHost(this, mHostName);
+}
+
+void Host::forceClose()
+{
+    if (!mpConsole) {
+        // The main console has gone away so we must already be dying
+        return;
+    }
+
+    mForcedClose = true;
+    postMessage(tr("[ ALERT ] - this profile is being forced to save and close."));
```

Comment:
I don't think we need additional user messages about this, Mudlet should be as least chatty as possible for the players.

@Mudlet/ui-review

Reply:
Well if one profile calls `closeMudlet()` than ALL the profiles get closed with no arguments - and if there is any sort of logging to file going on the user might want to record that is why the session terminated abruptly...

Reply:
I hear the explanation and now think this is valid then. 
However at least please upper the T after the dash starting a new sentence.

The "being forced" seems hard to grasp and hard to translate. How about reducing the cognitive load...
- ~This profile is being forced to save and close.~
- This profile is forced to save and close.
- This profile is asked to save and close.
- This profile was asked to save and close.
- This profile will now save and close.

Given we don't specify exactly "who" was the one being the force (and we shouldn't add that either), we can keep it shorter and simpler.
Either the player closed the profile herself (and we probably don't need a message then but hey) or it was asked by others and will now comply. Then it is okay to print that bit for future reference. If all goes well, it won't be displayed for long either. Only in log files it may accumulate for eternity, but it's small enough I guess.

```suggestion
    postMessage(tr("[ ALERT ] - This profile will now save and close."));
```

Reply:
That sounds great to me 👍